<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <script>
    async function main () {
      let p = new Promise((resolve, reject) => {
        // resolve('ok')
        reject('err')
      })
      // (1) await 右侧为 Promise 对象
      // let res = await p;
      // console.log(res); // ok

      // (2) await 右侧为 其他类型的数据
      // let res2 = await 20
      // console.log(res2); // 20

      // (3) 如果 await 的 Promise 失败了，会抛出异常，需要通过 try...catch 捕获处理
      try {
        let res3 = await p;
      } catch (e) {
        console.log(e); // err
      }
    }

    main()


  </script>
</body>

</html>